include/git2/sys/stream.h

```diff
@@ -39,6 +39,19 @@ typedef struct git_stream {
 	void (*free)(struct git_stream *);
 } git_stream;
 
+typedef int (*git_stream_ctor)(git_stream **out, const char *host, const char *port);
```

Comment:
I think that this might slightly better match some of the existing type names if it were `git_stream_cb`.  (See for example `git_transport_cb` which has a similar purpose.)